File: codeforces/32.hulk/hulk.cpp
#include <iostream>
#include <cstring>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned short int n;
  string hate = "I hate";
  string love = "I love";
  string feeling = hate;
  cin >> n;
  if (n > 1) {
    for (unsigned short int i = 2; i <= n; i++) {
      feeling += " that ";
      if ((i % 2) != 0) {
        feeling += hate;
      }
      else {
        feeling += love;
      }
    }
  }
  feeling += " it";
  cout << feeling << endl;
  return 0;
}

File: codeforces/32.hulk_[705A]/hulk.py
# Author: José Rodolfo (jric2002)
n = int(input())
hate = "I hate"
love = "I love"
feeling = hate
if (n > 1):
  for i in range(2, (n + 1)):
    feeling += " that "
    if ((i % 2) != 0):
      feeling += hate
    else:
      feeling += love
feeling += " it"
print(feeling)